ui/src/logger/logger.rs

```diff
@@ -0,0 +1,159 @@
+}
+
+#[derive(Debug, Clone)]
+pub enum LogLevel {
+    Debug,
+    Info,
+    Warn,
+    Error,
+}
+
+impl LogLevel {
+    pub fn from_str(log_level: &str) -> Self {
+        match log_level.trim() {
+            "Debug" => LogLevel::Debug,
+            "Warn" => LogLevel::Warn,
+            "Info" => LogLevel::Info,
+            "Error" => LogLevel::Error,
+            _ => LogLevel::Debug,
+        }
+    }
+
+    pub fn to_str(&self) -> &str {
+        match self {
+            LogLevel::Debug => "DEBUG",
+            LogLevel::Warn => "WARN",
+            LogLevel::Info => "INFO",
+            LogLevel::Error => "ERROR",
+        }
+    }
+
+    pub fn color(&self) -> &'static str {
+        match self {
+            LogLevel::Debug => "rgb(0, 255, 0)",
+            LogLevel::Info => "rgb(0, 195, 255)",
+            LogLevel::Warn => "yellow",
+            LogLevel::Error => "red",
+        }
+    }
+}
+
+pub struct Logger {
+    log_file: String,
+    log_entries: Mutex<Vec<Log>>,
+}
+
+impl Logger {
+    pub fn activate_logger() {
+        *LOG_ACTIVE.write() = true;
+    }
+
+    fn load() -> Logger {
+        let log_file = DEBUG_LOG_PATH.to_string();
+        let _ = OpenOptions::new().create(true).append(true).open(&log_file);
+        let log_entries = Mutex::new(Vec::new());
+        Logger {
+            log_file,
+            log_entries,
+        }
+    }
+}
+
+impl Logger {
+    fn log(&self, level: LogLevel, message: &str) {
+        if *LOG_ACTIVE.read() {
+            let new_log = Log {
+                level,
+                message: message.to_string(),
+                datetime: Local::now().to_string(),
+            };
+            let mut log_entries = self.log_entries.lock().unwrap();
+            log_entries.push(new_log.clone());
+
+            let mut file = OpenOptions::new()
+                .append(true)
+                .open(&self.log_file)
+                .unwrap();
+
+            if let Err(error) = writeln!(file, "{:?}", new_log.clone()) {
+                Logger::error(format!("Couldn't write to debug.log file. {error}").as_str());
+            }
+        }
+    }
+
+    pub fn debug(message: &str) {
+        LOGGER.read().log(LogLevel::Debug, message);
+    }
+
+    pub fn warn(message: &str) {
+        LOGGER.read().log(LogLevel::Warn, message);
+    }
+
+    pub fn info(message: &str) {
+        LOGGER.read().log(LogLevel::Info, message);
+    }
+
+    pub fn error(message: &str) {
+        LOGGER.read().log(LogLevel::Error, message);
+    }
+
+    pub fn show_log(&self) -> Vec<Log> {
+        let file = match File::open(DEBUG_LOG_PATH) {
+            Ok(log) => log,
+            Err(error) => {
+                Logger::error(format!("Unable to read debug.log file. {error}").as_str());
+                return Vec::new();
+            }
+        };
+
+        let reader = BufReader::new(file);
+        let mut logs: Vec<Log> = vec![];
+
+        let re_level = Regex::new(r#"level: (.*?),"#).unwrap();
```

Comment:
splitting by a delimiter could also work. though this would require the logs have a different format than relying on the `Debug` macro.